RAT/plotting.py

```diff
@@ -0,0 +1,188 @@
+
+                self._sld_plot.plot([row[0]-49 for row in new],
+                                    [row[1] for row in new],
+                                    color=self.color,
+                                    linewidth=1)
+
+        self._format_plots()
+
+        if self._delay:
+            plt.pause(0.005)
+
+    def _plot_shifted_data(self, sd, div):
+        """
+        Plots the shifted data.
+
+        Parameters
+        ----------
+        sd : nparray
+            The shifted data containing the x, y, e data
+        div : int
+            The divisor for the data
+        """
+        sd_x = sd[:, 0]
+        sd_y, sd_e = map(lambda x: x/div, (sd[:, 1], sd[:, 2]))
+
+        # Plot the errorbars
+        indices_removed = np.flip(np.nonzero(sd_y - sd_e < 0)[0])
+        sd_x_r, sd_y_r, sd_e_r = map(lambda x:
+                                     np.delete(x, indices_removed),
+                                     (sd_x, sd_y, sd_e))
+        self._plot_errorbars(sd_x_r, sd_y_r, sd_e_r, False)
+
+        # Plot one sided errorbars
+        indices_selected = [x for x in indices_removed
+                            if x not in np.nonzero(sd_y < 0)[0]]
+        sd_x_s, sd_y_s, sd_e_s = map(lambda x:
+                                     [x[i] for i in indices_selected],
+                                     (sd_x, sd_y, sd_e))
+        self._plot_errorbars(sd_x_s, sd_y_s, sd_e_s, True)
+
+    def _plot_errorbars(self, x, y, err, onesided):
+        """
+        Plots the error bars.
+
+        Parameters
+        ----------
+        x : nparray
+            The shifted data x axis data
+        y : nparray
+            The shifted data y axis data
+        err : nparray
+              The shifted data e data
+        onesided : bool
+            A boolean to indicate whether
+            to draw one sided errorbars
+        """
+        y_error = [[0]*len(err), err] if onesided else err
+        self._ref_plot.errorbar(x=x,
+                                y=y,
+                                yerr=y_error,
+                                fmt='none',
+                                ecolor=self.color,
+                                elinewidth=1,
+                                capsize=0)
+        self._ref_plot.scatter(x=x,
+                               y=y,
+                               s=3,
+                               marker="o",
+                               color=self.color)
+
+    def _format_plots(self):
+        """
+        Formats the ref and sld subplots.
+        """
+        self._ref_plot.set_yscale('log')
+        self._ref_plot.set_xscale('log')
+        self._ref_plot.set_xlabel('Qz')
+        self._ref_plot.set_ylabel('Ref')
+        self._ref_plot.legend()
+        self._ref_plot.grid()
+
+        self._sld_plot.set_xlabel('Z')
+        self._sld_plot.set_ylabel('SLD')
+        self._sld_plot.legend()
+        self._sld_plot.grid()
+
+    def wait_for_close(self):
```

Comment:
I think these functions can be tested easily, you might have to mock waitforbuttonpress. Its fine to just call the function directly as there is no way to easily emit the signals